File: 941-valid-mountain-array/valid-mountain-array.go
func validMountainArray(arr []int) bool {
    if len(arr) < 3 {
        return false;
    }
    i := 0;
    flagIncrease := false; // 上升
    flagDecrease := false; // 下降
    for ; i < len(arr)-1 && arr[i] < arr[i+1]; i++ {
        flagIncrease = true;
    }
    for ; i < len(arr)-1 && arr[i] > arr[i+1]; i++ {
        flagDecrease = true;
    }

    return i == len(arr)-1 && flagIncrease && flagDecrease; // i 有跑完整個數組且有上升跟下降 
}

File: 941-valid-mountain-array/valid-mountain-array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        if len(arr) < 3 : return False
        arr_len = len(arr)
        left, right = 0, arr_len - 1

        while left < arr_len - 1 and arr[left] < arr[left +1]:
            left += 1
        while right > 0 and arr[right] < arr[right -1] :
            right -= 1
        return right == left and right != 0 and left != (arr_len-1)